### ISSUE TITLE
[BUG] Issues #33 and #45 are back again

### ISSUE THREAD

**Issue by jefgenowitsch** [kroyee/SpeedBlocks#53]
I'm using the latest build and issue #33 and #45 are back again.
zDEFz thinks that a merge has overwritten the fixes.

**Comment by kroyee** [comment]
Nope, changes are still there. You are probably in the wrong branch locally.

**Comment by zDEFz** [comment]
@kroyee when i made the build yesterday i DELETED my old repository - cloned the master branch by getting a new fork and cloned that into my branch "name", overwritten gui.cpp with the fixed one and build it afterwards.
There is no possiblity that the branch "master" has it fixed.

The problem was really solved before came back after an update.
Or is it not fixed in branch **master**?

btw my gui.cpp is up to date containing for example

```
void UI::changeServerAdd(sf::String addr) { //Quickfix for issue #33
	std::string serveraddr = addr.toAnsiString();
	if (serveraddr.back() != '.' || serveraddr.size() != 0)
		net->serverAdd = addr.toAnsiString();
}
```

If a issue was fixed before and appears again then another change did break it.

**Whatever it is not fixed if i compile the latest branch master of you without changing anything**

**Comment by kroyee** [comment]
I don't know how your git and VS setup is structured, but the fix for the IP-address field i clearly there in your gui.cpp. Same goes for the combo-timer in gamePlay.cpp.
Are you building the right files with VS?
You said something about the git folder and the VS project folder not being the same and you need to copy the files between, so did you copy the updated files into your VS project?

All I know is that the fixes are still there in the repo. I can see them locally, and here on GitHub. Nothing has been deleted/changed.

**Comment by zDEFz** [comment]
Please let me clarify one thing!

I only use Visual Studio for Debugging purposes and have my git folder seperately!
So i grab the latest gui.cpp and update it each time.

Building is done first with visual studio and then with codeblocks to avoid false results.

**Comment by kb1900** [comment]
i can confirm that for the latest build #33 is still broken

simply dragging to select the text in the network ip input box causes the app to hang

**Comment by kroyee** [comment]
So it hangs without you even typing anything in it? Just by selecting?

**Comment by jefgenowitsch** [comment]
Yes, I can confirm it.

**Comment by kroyee** [comment]
Ok, adjusted the code again. This time it will hopefully stay away forever. :+1:

**Comment by zDEFz** [comment]
I can not pull anything correctly

@kroyee please let me in future just upload the file.

I can not pull properly. Github ..... ... No one can help.


No option did do.

**Comment by zDEFz** [comment]
fix for #33 partly.... removing does work but see by yourself

https://www.youtube.com/watch?v=obMlgO1xNbY&feature=youtu.be

**Comment by jefgenowitsch** [comment]
Issue #33 is still hanging for me.
Issue #45 is also not fixed.
I'm running the latest build.

**Comment by kroyee** [comment]
Are you sure about that?
#45 has been working this whole time for me. Just tested now, still working.

**Comment by zDEFz** [comment]
#33 new approach does work - however if delete the field and writing any character let it hang for some seconds IF no "." is before the character

darn.site = will hang
.darn.site = will NOT hang

**Comment by kroyee** [comment]
Ok.
I changed the approach all together, so the actual server address is not changed until you actually connect to the server (instead of when you edit the field).
If it still hangs, there must be something else altogether causing it in the GUI library.

I tested this on my Windows VM and the problem is fixed there after the new approach.

**Comment by zDEFz** [comment]
Nice work the update of today somehow fixed #33 completely.
Even if there was a letter in first place!
